Week1-Coding-Task.hs

```diff
@@ -0,0 +1,76 @@
+--Section 1, credit card validation
+--Double value of every second digit from beginning of right 
+--Add digits of doubled and undoubled values to form new number 
+--Calculate result mod 10: if 0, valid 
+
+------------------------------------------------------------------------------
+--Exercise 1: toDigits and toDigitsRev 
+------------------------------------------------------------------------------
+toDigits :: Integer -> [Integer]
+toDigitsRev :: Integer -> [Integer]
+
+--A list with elements reversed, so can use it later for reversing toDigits 
+rev :: [Integer] -> [Integer]
+rev [] = []--base case 
+rev (s:l) = rev l ++ [s]--step case
+
+toDigits s 
+    | s <= 0 = []
+    | s < 10 = [s] -- They have only one digit, so one element list 
+    | otherwise = toDigits (s`div`10) ++ [s`mod`10]--mod10 results only give one digit  
+
+toDigitsRev s = rev (toDigits s)--using rev here to raise abstraction
+
+------------------------------------------------------------------------------
+--Exercise 2: doubleEveryOther
+------------------------------------------------------------------------------
+--Length of a list 
+len :: [Integer] -> Integer
+len [] = 0--base case 
+len (_:l) = 1 + len l--step case
+
+doubleEveryOther :: [Integer] -> [Integer]
+doubleEveryOther [] = [] --Base case, empty list unaffected 
+doubleEveryOther (s:[]) = [s]--Base case, single element lists also unaffected 
+doubleEveryOther (t:(s:l)) 
+    | (len l)`mod`2 == 0 = [2*t, s] ++ doubleEveryOther l 
+    | otherwise          = [t, 2*s] ++ doubleEveryOther l
+
+------------------------------------------------------------------------------
+--Exercise 3: sumDigits 
+------------------------------------------------------------------------------
+--Uses toDigits function created before, with extra sum function 
+--Sum of a list  
+total :: [Integer] -> Integer
+total [] = 0--base case
+total (s:l) = s + total l--step case
+sumDigits :: [Integer] -> Integer
```

Comment:
Usually it's better to add empty line between functions. To make them separate visually.